aws-xray-recorder-sdk-core/src/test/java/com/amazonaws/xray/strategy/sampling/pollers/TargetPollerTest.java

```diff
@@ -1,32 +1,32 @@
 package com.amazonaws.xray.strategy.sampling.pollers;
 
-import com.amazonaws.services.xray.AWSXRayClient;
+import com.amazonaws.xray.internal.UnsignedXrayClient;
 import com.amazonaws.xray.strategy.sampling.manifest.CentralizedManifest;
+import java.time.Clock;
+import org.junit.Rule;
 import org.junit.Test;
-import org.junit.Assert;
-import org.mockito.Mockito;
-import org.powermock.reflect.Whitebox;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
 
-import java.time.Clock;
-import java.util.concurrent.ScheduledExecutorService;
+import static org.junit.Assert.assertTrue;
 
 public class TargetPollerTest {
```

Comment:
Can we add a `testPollerNotAbruptMainThread` for the target poller as well, for parity?

Reply:
I did the opposite, removing the test from `RulePoller`. I don't think these actually test anything. I noticed this comment

https://github.com/aws/aws-xray-sdk-java/blob/master/aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/strategy/sampling/pollers/TargetPoller.java#L43

`// An Error should not be handled by the application. The executor will die and not abrupt main thread.`

This comment doesn't make much sense to me - I'm not aware of any form of business logic that would cause the main thread to be interrupted from an executor (we're not passing the result back using e.g. `Future`) so I don't think it's possible for these tests to ever fail no matter what the code is. So they don't appear to test anything.

I don't know about executors dying either, even if a callback throws, the executor should still continue just fine.

Let me know if I'm missing anything.

Reply:
The test is indeed an invalid one. Executor would never abrupt the main thread.

I think the comment really should be
"The executor omits any exception except `Error`, which should stop the executor."

`ScheduledExecutor` dies if there's an unhandled exception. The below snippet will stop printing msg at the 5th execution
```
        AtomicInteger i = new AtomicInteger();
        Executors.newSingleThreadScheduledExecutor().scheduleAtFixedRate(() -> {
            if(i.incrementAndGet() == 5) {
                throw new IllegalStateException("STOP");
            }
            System.out.println("Hello World");
        }, 0, 1000, TimeUnit.MILLISECONDS);
```

Reply:
Oops - thanks for clarifying the `ExecutorService` behavior, never noticed that!

Tweaked the wording of the comment. Believe the behavior should be tweaked a bit to follow how I've seen most libraries propagate fatal errors (`AssertionError` is also `Error` but definitely not fatal...) but will look at that in another PR.

Reply:
Good call @shengxil, LGTM @anuraaga